src/Sarif/Writers/ResultLogJsonWriter.cs

```diff
@@ -241,13 +240,14 @@ public void WriteRules(IDictionary<string, IRule> rules)
             EnsureResultsArrayIsNotOpen();
             EnsureStateNotAlreadySet(Conditions.Disposed | Conditions.RulesWritten);
 
-            _jsonWriter.WritePropertyName("resources");
-            _jsonWriter.WriteStartObject(); // Begin: resources
 
-            _jsonWriter.WritePropertyName("rules");
-            _serializer.Serialize(_jsonWriter, rules, typeof(Dictionary<string, IRule>));
```

Comment:
>IRule [](start = 80, length = 5)

If you had changed this to `Rule` (because in fact `resources.Rules` is a `Dictionary<string,Rule>), would the rest of the `Rule`-to-`IRule` changes have been needed?